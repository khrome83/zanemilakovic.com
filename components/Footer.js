// @ts-check --jsx

import Section from '../components/Section';
import P from '../components/P';
import { GitHub, GitLab, LinkedIn, CodePen, Twitter } from '../components/Social';

const Footer = () => (
  <Section>
    <div className="footer">
      <div className="social">
        <P>
          <GitHub href="https://github.com/khrome83" />
          <GitLab href="https://gitlab.com/khrome83" />
          <CodePen href="https://codepen.io/khrome83/" />
          <LinkedIn href="https://www.linkedin.com/in/zanecmilakovic/" />
          <Twitter href="https://twitter.com/Khrome83" />
        </P>
      </div>
      <div className="copyright">
        <P>
          A Crafted Experience<br />
          by Zane Milakovic Â© 2017
        </P>
      </div>
      <style jsx>{`
        @import './styles/variables.css';

        .footer {
          text-align: center;
        }

        @media (--tablet-viewport) {
          .footer {
            display: flex;
            text-align: initial;
            align-items: center;
          }

          .copyright,
          .social {
            flex: 1;
          }

          .social {
            order: 1;
            text-align: end;
          }
        }
      `}</style>
    </div>
  </Section>
);

export { Footer as default };
