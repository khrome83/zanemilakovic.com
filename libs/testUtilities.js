// @ts-check --jsx
/* eslint-disable no-param-reassign */

const cleanSnapshots = (node) => {
  for (const prop in node) {
    if (prop === 'className') {
      const className = node[prop].replace(/jsx-[0-9]+[\s]?/g, '');
      node.className = className || '';
      if (!node.className) {
        delete node.className;
      }
    } else if (typeof node[prop] === 'object') {
      cleanSnapshots(node[prop]);
    }
  }
};

const lib = {
  cleanSnapshots,
};

export { lib as default, cleanSnapshots };
