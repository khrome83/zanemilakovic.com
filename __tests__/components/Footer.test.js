import React from 'react';
import renderer from 'react-test-renderer';
import { cleanSnapshots } from '../../libs/testUtilities';
import Footer from '../../components/Footer';

test('Footer renders correctly', () => {
  const component = renderer.create(<Footer />);
  const tree = component.toJSON();
  cleanSnapshots(tree);
  expect(tree).toMatchSnapshot();
});
