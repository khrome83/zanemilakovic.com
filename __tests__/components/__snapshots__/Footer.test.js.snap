// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<section
  className="section footer false"
>
  <div
    className="content"
  >
    <p
      className="p null"
    >
      A Crafted Experience by Zane Milakovic Â© 2017
    </p>
  </div>
</section>
`;
